dbms/src/Functions/FunctionsExternalDictionaries.h

```diff
@@ -1547,4 +1452,42 @@ class FunctionDictIsIn final : public IFunction
     const ExternalDictionaries & dictionaries;
 };
 
+
+template <typename T>
+auto getColumnData(const IColumn * column, size_t index) ->
+    std::enable_if_t<std::is_signed_v<T> && std::is_integral_v<T>, Int64>
+{
+    return column->getInt(index);
+}
+
+template <typename T>
+auto getColumnData(const IColumn * column, size_t index) ->
+    std::enable_if_t<std::is_unsigned_v<T> && std::is_integral_v<T>, UInt64>
+{
+    return column->getUInt(index);
+}
```

Comment:
This is Ok.

BTW it's also Ok to always use `getUInt` and convert to T.
Even if it's implementation defined, we can assume two's complement arithmetic.